homeassistant/components/sensor/rfxtrx.py

```diff
@@ -110,9 +111,9 @@ def __str__(self):
     @property
     def state(self):
         """Return the state of the sensor."""
-        if self.data_type:
+        if self.event:
             return self.event.values[self.data_type]
-        return None
+        return STATE_UNKNOWN
```

Comment:
I think there's a point in keeping this logic in one place: you don't need to worry in the entity class about which state to set before the first state report, less lines of code, less risk of setting the wrong state.

I don't understand what you mean: "But before the method never returned None". Which method and before when?